### PR TITLE
Update documentation.

### PR DESCRIPTION
**Pull request by csukuangfj**
Documentation can be written in either `*.md` or in `*.rst` format.
The `installation.md` is copied from INSTALLATION.md

### CODE REVIEW

**Comment by csukuangfj** [comment]
You can find what the generated documentation of this pullrequest looks like by visiting

https://k3.readthedocs.io/en/latest/

File: k2/docs/source/_k2.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# this file is for generating documentation
+
```
Comment (danpovey) [inline k2/docs/source/_k2.py:2]: So do the inclusion of these things make it document them?   Or not document them?

File: k2/docs/source/conf.py [hunk 2]
```diff
@@ -12,6 +12,8 @@
 #
```
Comment (danpovey) [inline k2/docs/source/conf.py:12]: How about we move the docs to (top-level)/docs/, and remove the old docs?

File: k2/docs/source/_k2.py (hunk 1 above)
Comment (csukuangfj) [reply k2/docs/source/_k2.py:2]: No, the purpose is to make `import k2` working as `k2` depends on `_k2`. The code is not going to be executed.
Everything that is going to be generated is specified explicitly in
https://github.com/k2-fsa/k2/blob/master/k2/docs/source/python_api/index.rst

```rst
Python API Reference
====================

.. toctree::
   :maxdepth: 2
   :caption: Python API

   fsa
   fsa_algo
   symbol_table
```

If we want to document some of the functions from `_k2`, we can document them here, add a file `python_api/_k2.rst`, and add it to `python_api/index.rst`.

File: k2/docs/source/conf.py (hunk 2 above)
Comment (csukuangfj) [reply k2/docs/source/conf.py:12]: Sure, will move it in the next pullrequest.

**Pull request merged**